scripts/log-search

```diff
@@ -0,0 +1,251 @@
+        "-n",
+        help="Number of log files to search",
+        choices=range(1, 16),
+        type=int,
+    )
+    log_selection_options.add_argument(
+        "--all-logs",
+        "-A",
+        help="Parse all logfiles, not just most recent",
+        action="store_true",
+    )
+
+    filtering = parser.add_argument_group("Filtering")
+    filtering.add_argument("filter", help="IP address or hostname to search for")
+    filtering.add_argument(
+        "--all-lines",
+        "-L",
+        help="Show all matching lines; equivalent to -suemtpr",
+        action="store_true",
+    )
+    filtering.add_argument("--static", "-s", help="Include static file paths", action="store_true")
+    filtering.add_argument("--uploads", "-u", help="Include file upload paths", action="store_true")
+    filtering.add_argument("--events", "-e", help="Include event fetch paths", action="store_true")
+    filtering.add_argument("--messages", "-m", help="Include message paths", action="store_true")
+    filtering.add_argument(
+        "--typing",
+        "-t",
+        help="Include typing notification path",
+        action="store_true",
+    )
+    filtering.add_argument("--presence", "-p", help="Include presence paths", action="store_true")
+    filtering.add_argument(
+        "--report", "-r", help="Include timing report paths", action="store_true"
+    )
+
+    output = parser.add_argument_group("Output")
+    output.add_argument("--full-line", "-F", help="Show full matching line", action="store_true")
+    return parser
+
+
+def maybe_gzip(logfile_name: str) -> TextIO:
+    if logfile_name.endswith(".gz"):
+        return gzip.open(logfile_name, "rt")
+    return open(logfile_name, "r")
+
+
+NGINX_LOG_LINE_RE = re.compile(
+    r"""
+      (?P<ip> \S+ ) \s+
+      - \s+
+      (?P<user> \S+ ) \s+
+      \[
+         (?P<date> \d+/\w+/\d+ )
+         :
+         (?P<time> \d+:\d+:\d+ )
+         \s+ \+0000
+      \] \s+
+      "
+         (?P<method> \S+ )
+         \s+
+         (?P<path> [^"]+ )
+         \s+
+         (?P<http_version> HTTP/[^"]+ )
+      " \s+
+      (?P<code> \d+ ) \s+
+      (?P<bytes> \d+ ) \s+
+      "(?P<referer> [^"]* )" \s+
+      "(?P<user_agent> [^"]* )" \s+
+      (?P<hostname> \S+ ) \s+
+      (?P<duration> \S+ )
+    """,
+    re.X,
+)
+
+
+class FilterType(Enum):
+    HOSTNAME = auto()
+    CLIENT_IP = auto()
+
+
+def main() -> None:
+    args = parser().parse_args()
+
+    logfile_names = ["/var/log/nginx/access.log"]
+    if args.all_logs:
+        logfile_count = 15
+    elif args.log_files is not None:
+        logfile_count = args.log_files
+    else:
+        # Detect if there was a logfile rotation in the last 3-ish
+        # hours, and if so include the previous logfile as well.
+        logfile_count = 1
+        try:
+            current_size = os.path.getsize(logfile_names[0])
+            past_size = os.path.getsize(logfile_names[0] + ".1")
+            if current_size < (3.0 / 24.0) * past_size:
```

Comment:
Do we want to make this 3 hours a parameter? Would help readability if nothing else.